lib/exercises.rb

```diff
@@ -1,19 +1,66 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: n*mlog(m) where m is the size of the words and n is the number of words. nm2 (nm squared) is the worst case. (I am confused here)
+# Space Complexity: O(n) for the making of the hash and return value
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  
+  sortedWords = {}
+
+  def abc(word)
+    word = word.chars.sort.join
+    puts word
+    return word
+  end
+
+  strings.each do |word|
+    abcWord = abc(word)
+
+    if !sortedWords[abcWord]
+      sortedWords[abcWord] = [word]
+    else 
+      sortedWords[abcWord].push(word)
+    end
+  end
+
+  return sortedWords.values
+
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity:  o(n2) there is a loop within a loop
+# Space Complexity: O(n) do to the new hash and arrray.
 def top_k_frequent_elements(list, k)
```

Comment:
👍 , however it could be more efficient to find the frequent items, maybe using the fact that you sorted them.

BTW did  you know that the `.max` method can take a parameter and return the `k` most max elements.